### DISCUSSION TITLE
MLKit Barcode Scanning - Accuracy

### DISCUSSION THREAD

**Discussion by randy-johnson** [capawesome-team/capacitor-mlkit#123] [Q&A] [answered]
Hello All,

I am using the capacitor-mlkit in an app I am creating for scanning code39 barcodes.

I am having trouble getting accurate scans using the plugin.

I have tried a few different barcode apps to test my barcodes and the other apps are getting accurate scans.

I have been through the plugin docs, but I am not seeing anything that I could change to help the accuracy.

I am looking for some advice on how to proceed.

Also I have tried this on IOS and Android and they are both suffering from the same issue

**Comment by randy-johnson** [comment]
fwiw, I gave https://github.com/capacitor-community/barcode-scanner a whirl this evening and it works every time.  

I also tried https://github.com/robingenz/capacitor-mlkit-plugin-demo wondering if perhaps it was something in my implementation and I had bad results there too.

**Comment by robingenz** [comment, marked as answer]
The scanned barcodes are processed by the ML Kit library, the plugin only serves as a wrapper. You should therefore report your problem in https://github.com/googlesamples/mlkit. There are [tips to improve the real-time performance](https://developers.google.com/ml-kit/vision/barcode-scanning/ios#performance-tips) but this only affects the real-time performance. If you also have the problem with the [`readBarcodesFromImage(...)`](https://capawesome.io/plugins/mlkit/barcode-scanning/#readbarcodesfromimage) method, then we cannot do anything about it.

> fwiw, I gave [capacitor-community/barcode-scanner](https://github.com/capacitor-community/barcode-scanner) a whirl this evening and it works every time.

You get different results with `capacitor-community/barcode-scanner`, as a different library is used there to process the barcodes.

**Reply by randy-johnson** [reply]
Hi Robin, ah yes, that makes sense.  I will reach out there.  

> If you also have the problem with the [readBarcodesFromImage(...)](https://capawesome.io/plugins/mlkit/barcode-scanning/#readbarcodesfromimage) method, then we cannot do anything about it.

Are you suggesting that I try using readBarcodesFromImage() to see if the problem exists there too?

Randy

**Reply by robingenz** [reply]
> Are you suggesting that I try using readBarcodesFromImage() to see if the problem exists there too?

Yes, exactly.
